gaphor/SysML/propertypages.py

```diff
@@ -28,53 +29,63 @@
 new_builder = new_resource_builder("gaphor.SysML")
 
 
-@PropertyPages.register(sysml.Requirement)
+@PropertyPages.register(RequirementItem)
 class RequirementPropertyPage(PropertyPageBase):
     order = 15
 
-    def __init__(self, subject: sysml.Requirement):
+    def __init__(self, item):
         super().__init__()
-        assert subject
-        self.subject = subject
-        self.watcher = subject.watcher()
+        self.item = item
 
     def construct(self):
+        if not self.item.subject:
+            return
+
+        subject = self.item.subject
+        watcher = subject.watcher()
+
         builder = new_builder(
             "requirement-editor",
             "requirement-text-buffer",
             signals={
+                "show-requirement-text-changed": (self._on_show_text_change,),
                 "requirement-id-changed": (self._on_id_changed,),
             },
         )
-        subject = self.subject
-
         entry = builder.get_object("requirement-id")
-        entry.set_text(subject.externalId or "")
+        entry.set_text(self.item.subject.externalId or "")
 
         text_view = builder.get_object("requirement-text")
 
         buffer = builder.get_object("requirement-text-buffer")
         if subject.text:
             buffer.set_text(subject.text)
 
+        show_requirement_text = builder.get_object("show-requirement-text")
+        show_requirement_text.set_active(self.item.show_text)
+
         @handler_blocking(buffer, "changed", self._on_text_changed)
         def text_handler(event):
             if not text_view.props.has_focus:
                 buffer.set_text(event.new_value)
 
-        self.watcher.watch("text", text_handler)
+        watcher.watch("text", text_handler)
 
         return unsubscribe_all_on_destroy(
-            builder.get_object("requirement-editor"), self.watcher
+            builder.get_object("requirement-editor"), watcher
         )
 
     @transactional
     def _on_id_changed(self, entry):
-        self.subject.externalId = entry.get_text()
+        self.item.subject.externalId = entry.get_text()
+
+    @transactional
+    def _on_show_text_change(self, button, gparam):
```

Comment:
removing qparam from the input variables caused an unhandled exception in the app. This was however not caught by any unit test. I'll think about a way to capture the scenario that let the exception.

Reply:
I see, I'll read more into this later